src/main/kotlin/org/rust/ide/intentions/RemoveMutableIntention.kt

```diff
@@ -0,0 +1,35 @@
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RsLetDecl
+import org.rust.lang.core.psi.RsPsiFactory
+import org.rust.lang.core.psi.util.parentOfType
+
+open class RemoveMutableIntention : RsElementBaseIntentionAction<RemoveMutableIntention.Context>(){
+    override fun getText() = "Remove mutable"
+    override fun getFamilyName() = text
+
+    open val mutable = true
+
+    override fun invoke(project: Project, editor: Editor, ctx: Context) {
+        val newLetExpr = RsPsiFactory(project).createLetDeclaration(ctx.letExpr.pat!!.firstChild.lastChild.text, ctx.letExpr.expr!!, !mutable, ctx.letExpr.typeReference)
+        ctx.letExpr.replace(newLetExpr)
+    }
+
+    protected fun isMutable(letExpr: RsLetDecl): Boolean {
+        val pat = letExpr.pat?.firstChild?.firstChild ?: return false
```

Comment:
I think you want to exactly switch on the type of possible patterns. Remember, you can have stuff like `let (foo, bar) = (1, 2)`.